Source/drlg_l2.cpp

```diff
@@ -1723,50 +1727,40 @@ bool PlaceMiniSet(const BYTE *miniset, int tmin, int tmax, int cx, int cy, bool
 	return true;
 }
 
-void PlaceMiniSetRandom(const BYTE *miniset, int rndper)
+void PlaceMiniSetRandom(const Miniset &miniset, int rndper)
 {
-	int sw = miniset[0];
-	int sh = miniset[1];
+	int sw = miniset.size.width;
+	int sh = miniset.size.height;
 
 	for (int sy = 0; sy < DMAXY - sh; sy++) {
 		for (int sx = 0; sx < DMAXX - sw; sx++) {
-			bool found = true;
-			int ii = 2;
-			if (sx >= nSx1 && sx <= nSx2 && sy >= nSy1 && sy <= nSy2) {
-				found = false;
-			}
-			for (int yy = 0; yy < sh && found; yy++) {
-				for (int xx = 0; xx < sw && found; xx++) {
-					if (miniset[ii] != 0 && dungeon[xx + sx][yy + sy] != miniset[ii]) {
-						found = false;
-					}
-					if (dflags[xx + sx][yy + sy] != 0) {
-						found = false;
-					}
-					ii++;
+			if (sx >= nSx1 && sx <= nSx2 && sy >= nSy1 && sy <= nSy2)
+				goto next;

```

Comment:
```suggestion
				continue;
```
No need to jump to the end of the current loop only to start the next iteration.

Reply:
I simply did it because there were other instances of `goto next` in the function. Let's see if @AJenbo likes gotos at all.

Reply:
I do not, we do not have any `goto` elsewhere in the code except for what we inherited 8 we inherited, 7 of them should be replaced with RAII and the last one is a copy of an SDL function that we keep for comparability with older versions of SDL and those do not want to modify (so that we can easily update it from upstream).

> I simply did it because there were other instances of `goto`

This is basically the problem with introducing them in the first place. They tend to breed and makes the code harder to read very quickly. Each instance is custom logic where `continue` is well defined.

Reply:
> They tend to breed

I guess `goto`s have a pretty good sex life. Let's not be envious of them.